### PR TITLE
Swap 1 Previous Owners and add 2 new Owners to `FakeFoundationOperationSafe`.

### PR DESCRIPTION
**Pull request by Ethnical**
**Description**

This PR will swap 1 old owner of the `FakeOperationSafe` on Sepolia using the function `swapOwners()` to remove 1 old signers (See the `README.md`).
And will add 2 owners using the function `addOwnerWithThreshold(address owner, uint256 _threshold)`. 
To do realize this , the task  `021-refresh-signers-FakeFoS` has been created under `sep` folder. 

**Recap** 

Using `swapOwners()`: 
This action will perform a swapping of the previous owner that was not active anymore by the new owner.

| PreviousOwner | newOwner |
| --- | --- |
| 0xad70Ad7Ac30Cee75EB9638D377EACD8DfDfE0C3c | 0x95E774787A63f145f7B05028a1479bDc9D055f3d |

Then using `addOwnerWithThreshold(address owner, uint256 _threshold)` we add 2 owners using function of the safe without changing the threshold.

To Recap the action will be:
| Owner | Threshold |
| --- | --- |
| 0x41fb1d8c3262e88a056ee3099f5718405CC8cAdE | 2 |
| 0xa03dafade71f1544f4b0120145eec9b89105951f| 2 |

**Tests**
I have run the simulation under tenderly and locally to make sure this is the desired state changes: 
**Tenderly:**
image

**Forge superchain-ops script**
image

**Additional context**
This safes doesn't implement any liveness module, imo this shouldn't trigger any fallback signers. 

**Metadata**
- Fixes https://github.com/ethereum-optimism/safe-wallets/pull/13

### CODE REVIEW

File: tasks/sep/021-refresh-signers-fakeFoS/SignFromJson.s.sol [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import "@eth-optimism-bedrock/src/dispute/lib/Types.sol";
+import {ModuleManager} from "safe-contracts/base/ModuleManager.sol";
+
+contract SignFromJson is OriginalSignFromJson {
+    using LibString for string;
+    // Safe contract for this task.
+    GnosisSafe foundationOperationsSafe =
+        GnosisSafe(payable(vm.envAddress("OWNER_SAFE"))); // We take from the "OWNER_SAFE" as this is the "TARGET_SAFE".
+
+    // TODO: Get the livenessGuard from the SC for not hardcoding the address.
```
Comment (blmalone) [inline tasks/sep/021-refresh-signers-fakeFoS/SignFromJson.s.sol:22]: Do you still need this TODO?

File: tasks/sep/021-refresh-signers-fakeFoS/SignFromJson.s.sol [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    function getAllowedStorageAccess()
+        internal
+        view
+        override
+        returns (address[] memory allowed)
+    {
+        allowed = new address[](2);
+        allowed[0] = address(foundationOperationsSafe);
+        allowed[1] = livenessGuard;
+        // allowed[2] = address(0xad70Ad7Ac30Cee75EB9638D377EACD8DfDfE0C3c);
```
Comment (blmalone) [inline tasks/sep/021-refresh-signers-fakeFoS/SignFromJson.s.sol:63]: Can these be deleted?

File: tasks/sep/021-refresh-signers-fakeFoS/SignFromJson.s.sol [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            foundationOperationsSafeOwners.length + 1 // 3 is the number of the address we wants to remove here.
+        );
+
+        for (uint256 i = 0; i < foundationOperationsSafeOwners.length; i++) {
+            shouldHaveCodeExceptions[i] = foundationOperationsSafeOwners[i];
+        }
+        // add the exception of the address that has to be removed.
+        shouldHaveCodeExceptions[
+            foundationOperationsSafeOwners.length
+        ] = address(0xad70Ad7Ac30Cee75EB9638D377EACD8DfDfE0C3c);
```
Comment (blmalone) [inline tasks/sep/021-refresh-signers-fakeFoS/SignFromJson.s.sol:49]: Maybe worth storing as a global variable and then checking that it's been removed.

**Review by blmalone — COMMENTED** [review]
This LGTM, some comments. I know that this is only a sepolia change so I'm not sure if we require `VALIDATIONS.md` to be present or not. 

Sounds like we're doing 1 swap and 2 adds, meaning the resulting number of owners will now be 10 instead of 8. It would be good to get an invariant check in the SignFromJson.s.sol file for this.

File: tasks/sep/021-refresh-signers-fakeFoS/SignFromJson.s.sol [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    /// @notice Checks the correctness of the deployment
+    function _postCheck(
+        Vm.AccountAccess[] memory accesses,
+        Simulation.Payload memory /* simPayload */
+    ) internal view override {
+        console.log("Running post-deploy assertions");
+
+        checkStateDiff(accesses);
```
Comment (blmalone) [inline tasks/sep/021-refresh-signers-fakeFoS/SignFromJson.s.sol:74]: Checking the new length of the owners array is correct would be a nice to have here.

File: tasks/sep/021-refresh-signers-fakeFoS/SignFromJson.s.sol (hunk 1 above)
Comment (Ethnical) [reply tasks/sep/021-refresh-signers-fakeFoS/SignFromJson.s.sol]: No thanks! I have remove the livenessGuard also in this commit: 698804d27093d0f6c0c662c6a183493b36614b18

File: tasks/sep/021-refresh-signers-fakeFoS/SignFromJson.s.sol (hunk 2 above)
Comment (Ethnical) [reply tasks/sep/021-refresh-signers-fakeFoS/SignFromJson.s.sol]: Yes, I have addressed the changes in this commit: 698804d27093d0f6c0c662c6a183493b36614b18

File: tasks/sep/021-refresh-signers-fakeFoS/SignFromJson.s.sol (hunk 3 above)
Comment (Ethnical) [reply tasks/sep/021-refresh-signers-fakeFoS/SignFromJson.s.sol]: Good idea, I have did that in this commit: 907c254e032068931e78f4c9c257bcdfaf10442a

File: tasks/sep/021-refresh-signers-fakeFoS/SignFromJson.s.sol (hunk 4 above)
Comment (Ethnical) [reply tasks/sep/021-refresh-signers-fakeFoS/SignFromJson.s.sol:84]: Nice Idea! I have did this in commit: caaa1c45f042e8fee050d385df053fcfe897a2f7

**Comment by Ethnical** [comment]
> This LGTM, some comments. I know that this is only a sepolia change so I'm not sure if we require `VALIDATIONS.md` to be present or not.

I tried to avoid to make `VALIDATIONS.md` to win some times to be honest. 
But, for posterity and people checking our tasks to create new one is probably better to add it. 
I will try to make the necessary files for each PRs.

**Review by blmalone — APPROVED** [review]
LGTM, I think a VALIDATIONS.md file would be good here.

**Comment by Ethnical** [comment]
@blmalone I have added the `VALIDATION.md` based on the simulation of [here](https://dashboard.tenderly.co/ethnical/project/simulator/7f43e66d-3f8f-41ab-ab18-6b5584432916/state-diff)

**Review by blmalone — APPROVED** [review]
Awesome, lgtm!
